tests/suite/test_ts_tls_passthrough.py

```diff
@@ -0,0 +1,225 @@
+    wait_before_test,
+    create_items_from_yaml,
+    delete_items_from_yaml,
+    wait_until_all_pods_are_ready,
+    get_first_pod_name,
+)
+from suite.custom_resources_utils import (
+    read_custom_resource,
+    read_ts,
+    patch_ts,
+    delete_ts,
+    create_ts_from_yaml,
+    create_virtual_server_from_yaml,
+    delete_virtual_server,
+)
+from suite.yaml_utils import get_first_host_from_yaml
+from suite.ssl_utils import get_server_certificate_subject, create_sni_session
+from settings import TEST_DATA
+
+class TransportServerTlsSetup:
+    """
+    Encapsulate Transport Server details.
+
+    Attributes:
+        public_endpoint (object):
+        ts_resource (dict):
+        name (str):
+        namespace (str):
+        ts_host (str):
+    """
+
+    def __init__(self, public_endpoint: PublicEndpoint, ts_resource, name, namespace, ts_host):
+        self.public_endpoint = public_endpoint
+        self.ts_resource = ts_resource
+        self.name = name
+        self.namespace = namespace
+        self.ts_host = ts_host
+
+
+@pytest.fixture(scope="class")
+def transport_server_tls_passthrough_setup(
+    request, kube_apis, test_namespace, ingress_controller_endpoint
+) -> TransportServerTlsSetup:
+    """
+    Prepare Transport Server Example.
+
+    :param request: internal pytest fixture to parametrize this method
+    :param kube_apis: client apis
+    :param test_namespace: namespace for test resources
+    :param ingress_controller_endpoint: ip and port information
+    :return TransportServerTlsSetup:
+    """
+    print(
+        "------------------------- Deploy Transport Server with tls passthrough -----------------------------------"
+    )
+    # deploy secure_app
+    secure_app_file = f"{TEST_DATA}/{request.param['example']}/standard/secure-app.yaml"
+    create_items_from_yaml(kube_apis, secure_app_file, test_namespace)
+
+    # deploy transport server
+    transport_server_std_src = f"{TEST_DATA}/{request.param['example']}/standard/transport-server.yaml"
+    ts_resource = create_ts_from_yaml(
+        kube_apis.custom_objects, transport_server_std_src, test_namespace
+    )
+    ts_host = get_first_host_from_yaml(transport_server_std_src)
+    wait_until_all_pods_are_ready(kube_apis.v1, test_namespace)
+
+    def fin():
+        print("Clean up TransportServer and app:")
+        delete_ts(kube_apis.custom_objects, ts_resource, test_namespace)
+        delete_items_from_yaml(kube_apis, secure_app_file, test_namespace)
+
+    request.addfinalizer(fin)
+
+    return TransportServerTlsSetup(
+        ingress_controller_endpoint,
+        ts_resource,
+        ts_resource["metadata"]["name"],
+        test_namespace,
+        ts_host,
+    )
+
+
+@pytest.mark.ts
+@pytest.mark.parametrize(
+    "crd_ingress_controller, transport_server_tls_passthrough_setup",
+    [
+        (
+            {
+                "type": "complete",
+                "extra_args": [
+                    "-enable-custom-resources",
```

Comment:
since `-enable-custom-resources` is true by default, would it make sense to get rid of it here?